sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -2085,4 +2087,29 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils with Te
       }
     }
   }
+
+  test("SPARK-28431: prevent CSV datasource throw TextParsingException with large size message") {
+    withTempDir { dir =>
+      val maxCharsPerCol = 10000
+      val str = "a" * (maxCharsPerCol + 1)
+
+      val csvFile = new File(dir, "data.csv")
+      Files.write(
+        csvFile.toPath,
+        Seq(str).asJava,
+        StandardOpenOption.CREATE, StandardOpenOption.WRITE
+      )
+
+      val errMsg = intercept[TextParsingException] {
+        spark.read
+          .option("maxCharsPerColumn", maxCharsPerCol)
+          .csv(csvFile.getAbsolutePath)
+          .count()
+      }.getMessage
+
+      assert(errMsg.endsWith("=..." + "a" * CSVOptions.maxErrorContentLength),
```

Comment:
It's okay. We can just check if there's `...` in the error message. The exact behaviour ideally has to be tested in Univocity library at https://github.com/uniVocity/univocity-parsers/blob/f616d151b48150bc9cb98943f9b6f8353b704359/src/test/java/com/univocity/parsers/common/DataProcessingExceptionTest.java.

We just need to test if this feature in Univocity is correctly enabled or not.